src/test/java/org/mule/test/http/functional/BasicHttpTestCase.java

```diff
@@ -20,114 +21,34 @@
-import org.eclipse.jetty.server.Request;
-import org.eclipse.jetty.server.Server;
-import org.eclipse.jetty.server.handler.AbstractHandler;
-import org.junit.After;
-import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import ru.yandex.qatools.allure.annotations.Features;
 
-//TODO: MULE-9702 Remove once the tests are migrated.
 @Features(HTTP_EXTENSION)
-public class BasicHttpTestCase extends AbstractHttpTestCase {
+public class BasicHttpTestCase extends AbstractHttpRequestTestCase {
 
   @Rule
-  public DynamicPort clientPort = new DynamicPort("clientPort");
-  @Rule
   public DynamicPort serverPort = new DynamicPort("serverPort");
 
-  protected Server server;
-
-  protected String method;
-  protected String uri;
-  private String query;
-  private Map<String, String> headers = new HashMap<>();
-
   @Override
   protected String getConfigFile() {
     return "basic-http-config.xml";
   }
 
-  @Before
-  public void startServer() throws Exception {
-    server = createServer();
-    server.setHandler(createHandler(server));
-    server.start();
-  }
-
-  @After
-  public void stopServer() throws Exception {
-    if (server != null) {
-      server.stop();
-    }
-  }
-
-  protected Server createServer() {
-    Server server = new Server(clientPort.getNumber());
-    return server;
-  }
-
-  protected AbstractHandler createHandler(Server server) {
-    return new TestHandler();
-  }
-
-  private class TestHandler extends AbstractHandler {
-
-    @Override
-    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
-        throws IOException, ServletException {
-
-      handleRequest(baseRequest, request, response);
-
-      baseRequest.setHandled(true);
-    }
-  }
-
-  protected void handleRequest(Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
-    extractBaseRequestParts(baseRequest);
-    writeResponse(response);
-  }
-
-  protected void extractBaseRequestParts(Request baseRequest) throws IOException {
-    method = baseRequest.getMethod();
-    uri = baseRequest.getUri().getCompletePath();
-    query = baseRequest.getUri().getQuery();
-    Enumeration<String> headerNames = baseRequest.getHeaderNames();
-    while (headerNames.hasMoreElements()) {
-      String headerName = headerNames.nextElement();
-      headers.put(headerName, baseRequest.getHeader(headerName));
-    }
-  }
-
-  protected void writeResponse(HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-    response.setStatus(HttpServletResponse.SC_OK);
-    response.getWriter().print("WOW");
-  }
-
   @Test
   public void sendsRequest() throws Exception {
     Event response = flowRunner("client").withPayload("PEPE").run();
-    assertThat(response.getMessage().getPayload().getValue(), is("WOW"));
+    assertThat(response.getMessage().getPayload().getValue(), is(DEFAULT_RESPONSE));
```

Comment:
Yes, cause I'm now extending from the abstract HTTP request test case that has another handler to avoid all that duplicate jetty server code.